__test__/file-promise.test.js

```diff
@@ -0,0 +1,49 @@
+/* eslint-disable no-undef */
+'use strict';
+jest.mock('fs');
+
+// eslint-disable-next-line no-unused-vars
+const faker = require('faker');
+const files = require('../lib/file-promise.js');
+
+describe('File handler', ()=>{
+  it('properly writes an object to a file', () =>{
+    let obj = {foo:'bar'};
+    return files.write('test.json', obj)
+      .then(success=> {
+        expect(success).toBeDefined();
+        // return files.read('test');
+
+      });
+
+  });
+
+
+  it ('properly writes a JSON string to file', () => {
+    let obj = {foo: 'bar'};
+    let str = JSON.stringify(obj);
+    return files.write('test.json', str)
+      .then(success => {
+        expect(success).toBeTruthy();
+        return files.read('test.json');
+      })
+      .then (json => {
+        expect(json.foo).toEqual('bar');
+      })
+      .catch(e => {
+        expect(e).toBeDefined();
+      });
+  });
```

Comment:
In this test, you have conflicting `expect` statements, that will result in your test always passing: 

```
write to file: 
    expect that file is written
    expect that an object property equals 'bar'

catch errors
    expect error to exist // conflicts with the write to file expectations
```

This test is looking for a proper write to a file. If the code writes to a file correctly, there should be no caught errors. But in this test, we're saying that if we caught an error, we should still pass the test. Instead, for error handling you should write a separate test.

Reply:
here ,error catch functions has been removed and some correction has been done to pass test